catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/ResourceOperationsTest.java

```diff
@@ -0,0 +1,502 @@
+        frameworkProperties.setPolicyPlugins(new ArrayList<>());
+        frameworkProperties.setSourcePoller(mockPoller);
+        frameworkProperties.setCatalogProviders(Collections.singletonList((CatalogProvider) provider));
+        frameworkProperties.setPostResource(mockPostResourcePlugins);
+        frameworkProperties.setFederationStrategy(mockFederationStrategy);
+        frameworkProperties.setFilterBuilder(new GeotoolsFilterBuilder());
+        frameworkProperties.setPreIngest(new ArrayList<>());
+        frameworkProperties.setPostIngest(postIngestPlugins);
+        frameworkProperties.setPreQuery(new ArrayList<>());
+        frameworkProperties.setPostQuery(new ArrayList<>());
+        frameworkProperties.setPreResource(new ArrayList<>());
+        frameworkProperties.setPostResource(new ArrayList<>());
+        frameworkProperties.setQueryResponsePostProcessor(mock(QueryResponsePostProcessor.class));
+        frameworkProperties.setStorageProviders(Collections.singletonList(storageProvider));
+        frameworkProperties.setMimeTypeMapper(new MimeTypeMapperImpl(Collections.singletonList(
+                mimeTypeResolver)));
+        frameworkProperties.setMimeTypeToTransformerMapper(mimeTypeToTransformerMapper);
+        frameworkProperties.setValidationQueryFactory(new ValidationQueryFactory(new GeotoolsFilterAdapterImpl(),
+                new GeotoolsFilterBuilder()));
+    }
+
+    @Test
+    public void testGetResource()
+            throws ResourceNotFoundException, IOException, ResourceNotSupportedException,
+            FederationException, UnsupportedQueryException, DownloadException {
+
+        boolean isEnterprise = true;
+        fanoutEnabled = true;
+        String resourceName = "bob";
+
+        Result resultMock = mock(Result.class);
+
+        ArrayList<Result> mockResultList = new ArrayList<>();
+        mockResultList.add(resultMock);
+
+        when(resourceRequestMock.getAttributeValue()).thenReturn(testUri);
+        when(resourceRequestMock.getAttributeName()).thenReturn("resource-uri");
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenReturn(
+                queryResponseMock);
+        when(queryResponseMock.getResults()).thenReturn(mockResultList);
+        when(queryResponseMock.getResults()
+                .get(0)
+                .getMetacard()).thenReturn(metacard);
+
+        when(reliableResourceDownloadManagerMock.download(any(), any(), any())).thenReturn(
+                resourceResponseMock);
+
+        Resource resourceMock = mock(Resource.class);
+        when(resourceResponseMock.getResource()).thenReturn(resourceMock);
+
+        // resourceOperations.setQueryOperations(queryOperationsMock);
+        frameworkProperties.setReliableResourceDownloadManager(reliableResourceDownloadManagerMock);
+
+        resourceOperations.getResource(resourceRequestMock,
+                isEnterprise,
+                resourceName,
+                fanoutEnabled);
+
```

Comment:
It would be ideal to have assertions or verifications on the return value of `getResource()`. For example, you could check the properties of the `ResourceResponse` to make sure they are correctly set, or `verify` certain things like `frameworkProperties.getFederatedSources().get()` are being invoked properly to make sure each conditional branch is being reached.